doc/decisions/header_include.md

```diff
@@ -0,0 +1,116 @@
+>
+> ### Definitions:
+>
+> - h-char: any member of the source character set except the new-line character and `>`
+> - q-char: any member of the source character set except the new-line character and `"`
+
+In short, the whole search process of the `#include` mechansim is implementation-defined.
+However, because `#include` works (mostly) like a literal copy-paste, deciding between `""` and `<>` within a public header file can affect user code.
+
+## Assumptions
+
+- Modern Compilers work like [GCC](https://gcc.gnu.org/onlinedocs/cpp/Search-Path.html):
+  - `#include "[header]"` treats `[header]` as file path relative to the current file (as defined by the standard, if such a file doesn't exist there is a fallback to the `<>` behavior)
+  - `#include <[header]>` treats `[header]` as file path relative to one of the pre-defined include-paths
+
+## Considered Alternatives
+
+- Let developers decide:
+
+  This would just create inconsistent code.
+
+- Always use `<>`:
+
+  This can make things more difficult than necessary. We wouldn't utilize the common "relative path" behaviour. Therefore, user code must have all the same directories in the include-paths as our code (translated to their installed counterparts).
+
+- Put all headers into one directory and use the same relative layout as will be installed. Then use `""`:
+
+  This makes for an inconvenient development experience. It is also pretty hard to achieve for plugins and private headers would clutter the include directory.
+
+## Decision
+
+- Utilize CMake to make the following rules work.
+- Including another header from the current directory is done with:
+  ```c
+  #include "header.h"
+  ```
+- Including another header from a subdirectory of the current one (e.g. from a third-party library whose code was copied into the repo) is done with:
+  ```c
+  #include "subdir/header.h"
+  ```
+- Including a header from another library within the Elektra codebase, is done with:
+  ```c
+  #include "../otherlib/header.h"
+  ```
+- Including Elektra library headers from a plugin, is done with:
+  ```c
+  #include <elektra/thelib/header.h>
+  ```
+  In other words, plugins are treated as external and must use the same rules an application would also use.
+- Only including system headers and headers from external third-party libraries is done with:
+  ```c
+  #include <external.h>
+  ```
+
+## Rationale
+
+See also considered alternatives.
```

Comment:
To put everything under "elektra" has a clear rationale:

```suggestion
We avoid conflicts when header files with the same file name exist.
```

It could be clearer what the whole purpose of this decision is. Why do we want relative includes?

Reply:
I will clarify but the main reasons are:
1. With the new directory structure using relative imports is much easier and makes it clearer which headers are part of the same library.
2. I remember having many problems with mismatched include paths when working on the High-level API. If you use `<>` either the CMake setup gets complicated, or the user must pass the correct directories to their compiler. With `""` it works as long as relative paths are preserved during install.